### PR TITLE
[DUOS-1972][risk=no] Add sharing plan file name

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses

Partially addresses https://broadworkbench.atlassian.net/browse/DUOS-1972

Minor additions to the dataset so we can store alternate data sharing plan file location.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/models/Dataset.java [hunk 1]
```diff
@@ -287,6 +289,14 @@ public void setSharingPlanDocument(String sharingPlanDocument) {
         this.sharingPlanDocument = sharingPlanDocument;
     }
 
+    public String getSharingPlanDocumentName() {
+        return sharingPlanDocumentName;
+    }
+
+    public void setSharingPlanDocumentName(String sharingPlanDocumentName) {
+        this.sharingPlanDocumentName = sharingPlanDocumentName;
+    }
```
Comment (connorlbark) [inline src/main/java/org/broadinstitute/consent/http/models/Dataset.java:298]: should this be added to the dataset schema in swagger?

Comment (rushtong) [reply src/main/java/org/broadinstitute/consent/http/models/Dataset.java:298]: Yes, thank you!

**Review by quazi-h — APPROVED** [review]
Changes to add the sharing plan file name look good!

File: src/main/resources/changesets/changelog-consent-98.0.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+         http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+    <changeSet id="98.0" author="grushton">
+        <addColumn tableName="dataset">
+            <column name="sharing_plan_document_name" type="varchar(255)"/>
+        </addColumn>
+        <rollback>
+            <dropColumn tableName="dataset" columnName="sharing_plan_document_name"/>
+        </rollback>
```
Comment (connorlbark) [inline src/main/resources/changesets/changelog-consent-98.0.xml:10]: just curious (does not affect approval): is this added so we can turn back changes in changesets? should i be adding rollback sections to all of my changesets moving forward?

**Review by connorlbark — APPROVED** [review]
looks good!

File: src/main/resources/changesets/changelog-consent-98.0.xml (hunk 2 above)
Comment (rushtong) [reply src/main/resources/changesets/changelog-consent-98.0.xml:10]: This is the failure case. If anything goes wrong, this is the solution to the error. Yes, we should usually have rollbacks in the cases where things can be rolled back.

**Pull request merged**